### PR TITLE
split  get_param_by_action for mut/immutable option so that

### PR DESCRIPTION
**Pull request by dmytro-shash**
view! could work properly with get_entity by_token ( & self , —args )

refactored test_utils to work with tests indeed.

fixed decrease_borrows to be compatible with repay

### CODE REVIEW

**Comment by Pavel-Blaize** [comment]
Task linked: [CU-1z4fg4r Tests are broken on master](https://app.clickup.com/t/1z4fg4r)

File: contracts/controller/src/borrows_supplies.rs [hunk 1]
```diff
@@ -12,26 +12,28 @@ pub enum ActionType {
 #[near_bindgen]
 impl Contract {
     #[private]
-    fn set_entity_by_token(&mut self, action: ActionType, account: AccountId, token_address: AccountId, token_amount: Balance) -> Balance {
+    fn set_entity_by_token(&mut self, action: ActionType, account: AccountId, token_address: AccountId, tokens_amount: Balance) -> Balance {
```
Comment (markblaize) [inline contracts/controller/src/borrows_supplies.rs:15]: @dmytro-shash `tokens_amount` -> `tokens_amount`

File: contracts/controller/src/borrows_supplies.rs [hunk 2]
```diff
@@ -136,106 +132,92 @@ impl Contract {
...
 
+
         let _existing_supplies = self.get_entity_by_token(Supply, account.clone(), token_address.clone());
         // TODO add check if allowed  (USD-estimated ACCOUNT SUPPLIES > USD-estimated ACCOUNT BORROWED  * ratio ? (or just 0.8) )
 
         // FIXME mock-checking for now
         return true;
     }
-
-    pub fn make_borrow(
```
Comment (markblaize) [inline contracts/controller/src/borrows_supplies.rs:181]: @dmytro-shash why this method was deleted?

File: contracts/controller/src/lib.rs [hunk 3]
```diff
@@ -51,6 +50,9 @@ pub struct Contract {
 
     /// Contract configuration object
     pub config: LazyOption<Config>,
+
+    /// Contract admin account (controller itself by default)
+    pub admin: AccountId,
```
Comment (markblaize) [inline contracts/controller/src/lib.rs:55]: @dmytro-shash this change is out of scope of task

File: contracts/controller/src/test_utils.rs [hunk 4]
```diff
@@ -1,14 +0,0 @@
-// use near_sdk::AccountId;
```
Comment (markblaize) [inline contracts/controller/src/test_utils.rs:1]: @dmytro-shash I suppose this file should resolve problem with DRY for `init_test_env`. This was commented out because it's breaks compilation but should be fixed and correctly used inctead of duplicating `init_test_env` everywhere

**Review by markblaize — CHANGES_REQUESTED** [review]
